### ISSUE TITLE
Mobile query example (gmf) is broken

### ISSUE THREAD

**Issue by tsauerwein** [camptocamp/ngeo#663]
Example contribs/gmf/examples/mobilequery.html on current master is broken. When clicking on the map with the default theme "OSM", no request is made.

A first shot with the debugger tells me that the sources are [not in the cache](https://github.com/camptocamp/ngeo/blob/ba34127/src/services/query.js#L308) but I have no idea why.

**Comment by sbrunner** [comment]
Duplicate with #662?

**Comment by tsauerwein** [comment]
No, it's also for the non-hosted examples running under localhost:3000.

**Comment by adube** [comment]
I know what's going on.

First of all, there's one more issue in the mobile application.  The query cache is completely empty because the `gmf.QueryManager` is no longer included in the app controller.  I think it was removed by mistake.  It used to be there, see: https://github.com/camptocamp/ngeo/pull/633/files#diff-80f5226d8a311a69cb873167d904531fR4.  That populates the cache properly.

Then, there's the other issue which I'm pretty sure comes from the following change: 43f680c2.  It would seem that layers are now created with the combination of multiple nodes in order to have less layer objects created.  The way the query sources are created are one per leaf (nodes that have no children).  But the layer is created with the combination of all those leaves and use the parent node id as `querySourceId`.

To summarize: the parent node doesn't have a query source created and all current leaf nodes won't be used, ever.

I think I can come up with a fix.

**Comment by adube** [comment]
@ger-benjamin, I need to talk to you about this, as you're the one who re-factored the layer tree.  I'll call you tomorrow, but here's a summary of my analysis.

The `gmf.QueryManager` is currently hooked on the `gmf.Themes` service and is responsible of creating the query sources.  When it was 1 layer == 1 source config, that worked fine.  Now, with the layers created by combining multiple nodes, that won't work as much as well any more. 

A query source can either be created with a `ol.source.Source` object or be given the parameters for the query manager to create one itself.  With only one layer, we have only one source in which the layer names are added/removed.  Logically, the query manager would need to share that same source object, but it's currently created by the layer tree.

I think we shouldn't continue to let the manager create query sources without a direct reference to the source objects themselves.  If we don't, then we'll end up with having 2 sources that won't be synchronized with their layers.

I'll call you tomorrow about this.

**Comment by ger-benjamin** [comment]
> ... the `gmf.QueryManager` is no longer included in the app controller. ...

I'm including the "gmf.mobiledisplayqueries" in the mobile app now. It's currently on a branch on my fork of ngeo. I've already re-included the `gmf.QueryManager` in the abstract controller and work on the css. So, please, don't re-include it in another PR. (But really thanks for the tips, I think It was erased because it need to be injected but it's not used in the file where it's included... (seems useless but it is not)).

For the layertree:
Yep, now the layer tree create only layer on firsts nodes. The aim was to make only one request per "WMS group" but for some reason, I've also changed the manner that I handle external WMS and  WMTS. For these case, the layertree create a ol.layer.Group on each first node (and only here).

So each leaf is empty first (but knows its source node and its parent). For performance, a WMTS or external WMS layer can be linked in the leaf's "layer" parameter. But internal WMS will be ever located in the first group node.

**Comment by adube** [comment]
@ger-benjamin So, are you planning on solving the issue raised here ?

**Comment by ger-benjamin** [comment]
Not yet, for now, I include the queryexample "as is" in the mobile app.

**Comment by adube** [comment]
Okay.  So, I'll leave this in your hands for now.  Please, let me know if you wish to discuss this more or if you need any help.

**Comment by ger-benjamin** [comment]
Ref to: https://github.com/camptocamp/ngeo/pull/692

**Comment by ger-benjamin** [comment]
Close with  #692
